src/Avalonia.Visuals/Rendering/DeferredRenderer.cs

```diff
@@ -355,15 +355,19 @@ private void Render(IDrawingContextImpl context, VisualNode node, IVisual layer,
 
                     node.BeginRender(context, isLayerRoot);
 
-                    foreach (var operation in node.DrawOperations)
+                    var drawOperations = node.DrawOperations;
+                    for (int i = 0; i < drawOperations.Count; i++)
```

Comment:
If you want to make it even faster you can store `drawOperations.Count` into a local. Otherwise you will end up with extra `.Count` call per loop iteration.

Reply:
yep, I believe there is a plan for a renderer rewrite, stiil probably worth speeding up the obvious things